### DISCUSSION TITLE
Dynamic block cannot save and use InnerBlocks content

### DISCUSSION THREAD

**Discussion by stewardccm** [WordPress/gutenberg#44466] [Q&A] [answered]
### Description

I want to add InnerBlocks support in dynamic blocks.

I added InnerBlocks support in edit.js after creating dynamic blocks using @wordpress/create-block.

In the editor, any block can be added to the custom block, but it cannot be saved, and the render function shows empty content.

### Step-by-step reproduction instructions

1. Use @wordpress/create-block to create a dynamic block
2. Add InnerBlocks support in edit.js
3. Render InnerBlocks content in template.php
4. Open in theme editor and add the dynamic block with some InnerBlocks
5. Save the template
6. Refresh the editor
7. It show empty InnerBlocks in dynamic block

### Screenshots, screen recording, code snippet

![Screenshot](https://user-images.githubusercontent.com/5201752/192239837-5c975df9-7cbd-4269-85c6-fe84213bede5.gif)

![image](https://user-images.githubusercontent.com/5201752/192239988-694e0cc2-d5e9-4461-bffc-937bffab00bd.png)

![image](https://user-images.githubusercontent.com/5201752/192240204-32e4662b-0107-42c4-ab09-c28a1f7e282d.png)

![image](https://user-images.githubusercontent.com/5201752/192240247-d63df484-8fe4-42b0-9bee-5eb984f3324f.png)

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by t-hamano** [comment, marked as answer]
If you want to use the innerBlock in a dynamic block, I think you need to save the innerBlock with the save function.:
```javascript
import { InnerBlocks } from '@wordpress/block-editor';
export default function save() {
	return <InnerBlocks.Content />;
}
```

Also, I think this is a question, so I will convert it to a Discussion.

**Reply by stewardccm** [reply]
oh i thought this was a bug but thank you for your support
